### PR TITLE
fix(search): merge and aireraking are now accessible as json.RawMessage

### PR DESCRIPTION
**Pull request by vredens**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | yes
| New feature?      | no
| BC breaks?        | no     
| Related Issue     | 
| Need Doc update   | no

## Describe your change

Added two new properties to the search.QueryRes data structure (`Merge` and `AIReRanking`) so API response information under `merge` and `aiReRanking` is available.

## What problem is this fixing?

Query response data structure (`search.QueryRes`) does not include `merge` and `aiReRanking` information. This data is needed for debugging and observability related to the use of personalization and AI ReRanking.

### CODE REVIEW

File: algolia/search/responses_search.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 )
 
 type QueryRes struct {
+	AIReRanking                json.RawMessage                   `json:"aiReRanking"`
```
Comment (helenamariano) [inline algolia/search/responses_search.go:10]:
```suggestion
	AIReRanking               *AIReRanking                 `json:"aiReRanking"`
```

Comment (helenamariano) [reply algolia/search/responses_search.go:10]: We can directly marshal into the `AIReRanking` type

File: algolia/search/responses_search.go [hunk 2]
```diff
@@ -22,6 +23,7 @@ type QueryRes struct {
 	Index                      string                            `json:"index"`
 	IndexUsed                  string                            `json:"indexUsed"`
 	Length                     int                               `json:"length"`
+	Merge                      json.RawMessage                   `json:"merge"`
```
Comment (helenamariano) [inline algolia/search/responses_search.go:26]:
```suggestion
	Merge                      *Merge                  `json:"merge"`
```

File: algolia/search/responses_search.go [hunk 3]
```diff
@@ -91,6 +93,14 @@ func (r QueryRes) UnmarshalUserData(v interface{}) error {
 	return json.Unmarshal(userDataPayload, &v)
 }
 
+func (r QueryRes) UnmarshalMerge(v interface{}) error {
```
Comment (helenamariano) [inline algolia/search/responses_search.go:96]: We don't need this if we marshal directly into the object.

**Review by helenamariano — CHANGES_REQUESTED** [review]
Looks great! Just some small changes.

File: algolia/search/responses_merge.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	NBPersoSkipped     int                     `json:"nbPersoSkipped"`
+	Percentile         int                     `json:"percentile"`
+	NBRelevanceBuckets int                     `json:"nbRelevanceBuckets"`
+	Profile            *PersonalizationProfile `json:"profile"`
+}
+
+type PersonalizationProfile struct {
+	TaskID int64 `json:"taskID"`
+	Time   int64 `json:"time"`
+	Facets map[string]int
```
Comment (Fluf22) [inline algolia/search/responses_merge.go:29]:
```suggestion
	Facets map[string]int `json:"facets"`
```

**Comment by vredens** [comment]
All fixed. Thanks for the review @helenamariano and @Fluf22

**Review by Fluf22 — APPROVED** [review]
Thanks!
@helenamariano I'll let you give the final approval

**Review by helenamariano — APPROVED** [review]

**Pull request merged**